### PR TITLE
Add index wrapper around versioned javadocs to allow selecting different versions

### PR DESCRIPTION
**Pull request by bdemers**
versions.js will be generated at release time
index.html loads populates dropdown with content from versions.js

Dropdown will be populated similar to:
```
X.X.2 [Current]
X.X.1
-------------------
0.0.4 [Legacy]
```

`versions.js` will be generated at release time (from different branch, so different PR)
but basically to get the list of tags:
all tags: `$(git tag | grep okta-sdk-root- | sed 's_okta-sdk-root-__' | sort -r)`
latest tag: `$(git tag | grep okta-sdk-root- | sed 's_okta-sdk-root-__' | sort -r | head -1)`

### CODE REVIEW

**Comment by bdemers** [comment]
@robertjd  Some pretty raw JS.  This isn't critical, so take a look at it when you can.   The idea here is to make this standalone from developer.github.io, but if there is something I can use to remove some of this logic let me know

File: versions.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+window.apiVersions = {
+versions: [
+"0.8.1",
```
Comment (jmelberg-okta) [inline versions.js]: Each nested element should be separated by two spaces here.

```javascript
window.apiVersions = {
  "0.8.1",
  "0.8.0",
  ...
}
```

File: index.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// addSeparator();
+
+for (version of window.apiVersions.legacy) {
+  addVersion(version, version + ' [Legacy]');
+}
+
+$('#currentSelection').html(window.apiVersions.current + ' [Current] <span class="caret"></span>');
+
+$('#versionsList').append()
+window.apiVersions.current
```
Comment (jmelberg-okta) [inline index.html]: Is this supposed to be appended to the `versionsList`?

File: index.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+// addVersion('development', 'Development');
+// addSeparator();
+
+for (version of window.apiVersions.legacy) {
+  addVersion(version, version + ' [Legacy]');
+}
+
+$('#currentSelection').html(window.apiVersions.current + ' [Current] <span class="caret"></span>');
+
+$('#versionsList').append()
```
Comment (jmelberg-okta) [inline index.html]: Missing `;`

**Comment by bdemers** [comment]
Looks like the css 404 are related to this: https://bugs.openjdk.java.net/browse/JDK-8093199
I'll create a separate issue for that one, as they are unrelated

File: index.html [hunk 4]
```diff
@@ -50,8 +50,7 @@
 
 $('#currentSelection').html(window.apiVersions.current + ' [Current] <span class="caret"></span>');
 
-$('#versionsList').append()
-window.apiVersions.current
+$('#versionsList').append();
```
Comment (jmelberg-okta) [inline index.html]: Since nothing is being appended here, you can remove this line

Comment (bdemers) [reply index.html]: 👍

File: index.html [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        .attr('href', version + '/apidocs/index.html')
+        .attr('target', 'javadocs')
+        .text(versionName)
+    )
+  );
+}
+
+$(function() {
+
+  for (version of window.apiVersions.versions) {
```
Comment (robertjd) [inline index.html]: for.. of won't work in IE < Edge

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of#Browser_compatibility

File: index.html [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  versionName = versionName || version
+
+  $('#versionsList').append(
+    $('<li>').append(
+      $('<a>')
+        .attr('href', version + '/apidocs/index.html')
+        .attr('target', 'javadocs')
+        .text(versionName)
+    )
+  );
```
Comment (robertjd) [inline index.html]: nit: inconsistent semicolon usage, you can use eslint to help with this stuff

Comment (bdemers) [reply index.html]: Is there an _easy_ way to run eslint on a bare html file?

**Comment by bdemers** [comment]
@jmelberg-okta @robertjd take another look

File: index.html [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    $(this).parents(".dropdown").find('.dropdown-toggle').html($(this).text() + ' <span class="caret"></span>');
+    $(this).parents(".dropdown").find('.dropdown-toggle').val($(this).data('value'));
+  });
+});
+</script>
+
+<div class="fill">
+    <ol class="breadcrumb" style="margin-bottom: 0px;">
+        <li><a href="/"><img src="images/okta-dev-logo-48.png" style="height:1.7em"></a>
+            Okta Java SDK
```
Comment (jmelberg-okta) [inline index.html]: Maybe embed this in the `<a>` so it is clickable too?

File: versions.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+window.apiVersions = {
```
Comment (jmelberg-okta) [inline versions.js]: Needs extra indentation at the first level -> 
```javascript
window.apiVersions = {
  versions: [
    '0.8.1',
    '0.8.0',
    '0.7.0',
    '0.6.0',
    '0.5.0'
  ],
  legacy: [
    '0.0.4'
  ],
  current: '0.8.1'
};
```

File: index.html [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    }  
+    addVersion(version, versionName);
+  }
+
+  addSeparator();
+  // TODO: add dev version in the future
+  // addVersion('development', 'Development');
+  // addSeparator();
+
+  for (version of window.apiVersions.legacy) {
```
Comment (jmelberg-okta) [inline index.html]: As @robertdamphousse-okta mentioned, this doesn't work in IE. Follow the same logic as you did on line `34-39`

Comment (bdemers) [reply index.html]: doh

File: index.html (hunk 7 above)
Comment (bdemers) [reply index.html]: +1

File: index.html [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    addVersion(version, versionName);
+  }
+
+  addSeparator();
+  // TODO: add dev version in the future
+  // addVersion('development', 'Development');
+  // addSeparator();
+
+  for (let ii = 0; ii < window.apiVersions.legacy.length; ii++) {
+    let version =window.apiVersions.legacy[ii];
```
Comment (jmelberg-okta) [inline index.html]: `=window` -> `= window`

Comment (bdemers) [reply index.html]: easy one!

**Pull request merged**